bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -649,13 +752,15 @@ private synchronized void fatalError(Object listener, Http2Exception cause) {
      * delay in order to prevent sequencing issues.
      *
      * @param listener the entity that caused this method to be called.
-     * @param cause the exception that caused the error.
+     * @param cause the type of exception that caused the error.
+     * @param sessionId the identity of the session on which the error occurred.
+     * @param streamId the identity of the stream on which the error occurred.
      */
-    protected void fatalErrorDelayed(Object listener, Http2Exception cause) {
+    protected void fatalErrorDelayed(Object listener, Http2Exception cause, int sessionId, int streamId) {
```

Comment:
As above

Reply:
I think we can dispense with the streamId but we still need the sessionId (see above).

Reply:
I think we can pass session in `fatalErrorDelayed()` but the hold on the session MUST have been released before `fatalError()` gets called .. one second later.

Reply:
That would already be an improvement, but I'm not sure timing of GC actually destroying the session object matters that much. Anything related to the actual HTTP handling will be done by methods, not by the GC, anyway.

Reply:
FWIW I think the Jetty Session object implements AutoCloseable so the Jetty code should (perhaps) clean it up .. so long as nothing else is holding a reference to it. But anyway point taken. See revised code committed just now.